cmd/argo/commands/cron/create.go

```diff
@@ -0,0 +1,100 @@
+package cron
+
+import (
+	"github.com/argoproj/argo/workflow/templateresolution"
+	"log"
+	"os"
+
+	"github.com/argoproj/pkg/json"
+	"github.com/spf13/cobra"
+
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/workflow/common"
+	"github.com/argoproj/argo/workflow/util"
+	"github.com/argoproj/argo/workflow/validate"
+)
+
+type cliCreateOpts struct {
+	output string // --output
+	strict bool   // --strict
+}
+
+func NewCreateCommand() *cobra.Command {
+	var (
+		cliCreateOpts cliCreateOpts
+	)
+	var command = &cobra.Command{
+		Use:   "create FILE1 FILE2...",
+		Short: "create a cron workflow",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				cmd.HelpFunc()(cmd, args)
+				os.Exit(1)
+			}
+
+			CreateCronWorkflows(args, &cliCreateOpts)
+		},
+	}
+	command.Flags().StringVarP(&cliCreateOpts.output, "output", "o", "", "Output format. One of: name|json|yaml|wide")
+	command.Flags().BoolVar(&cliCreateOpts.strict, "strict", true, "perform strict workflow validation")
+	return command
+}
+
+func CreateCronWorkflows(filePaths []string, cliOpts *cliCreateOpts) {
+	if cliOpts == nil {
+		cliOpts = &cliCreateOpts{}
+	}
+	defaultCronWfClient := InitCronWorkflowClient()
+
+	fileContents, err := util.ReadManifest(filePaths...)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var cronWorkflows []wfv1.CronWorkflow
+	for _, body := range fileContents {
+		cronWfs := unmarshalCronWorkflows(body, cliOpts.strict)
+		cronWorkflows = append(cronWorkflows, cronWfs...)
+	}
+
+	if len(cronWorkflows) == 0 {
+		log.Println("No CronWorkflows found in given files")
```

Comment:
I think this probably ok?

Reply:
I've found this to be useful when accidentally submitting `Workflow`s instead of `CronWorkflow`s to `argo cron create`